src/mongoc/mongoc-bulk-operation.c

```diff
@@ -851,6 +851,16 @@ mongoc_bulk_operation_set_client (mongoc_bulk_operation_t *bulk, void *client)
 }
 
 
+void
+mongoc_bulk_operation_set_client_session (mongoc_bulk_operation_t *bulk,
+                                          void *client_session)
```

Comment:
mongoc_client_t *, not void *

Reply:
I ran into a host of compile errors attempting to use `mongoc_client_session_t *` here. I assume it has something to do with include order, as it's immediately caused by including `mongoc-client-session.h` up top.

Reply:
Don't include mongoc-client-session.h, and add a forward declaration:
```
/* forward decl */
struct _mongoc_client_session_t;

void
mongoc_bulk_operation_set_client_session (mongoc_bulk_operation_t *bulk,
                                          struct _mongoc_client_session_t *client_session)
```
See my solution to a similar problem here:

https://github.com/mongodb/mongo-c-driver/blob/0dc8fb1aca4194df67afd431c55d438f9ecef56c/src/mongoc/mongoc-apm-private.h#L30-L29

Preserves typechecking, solves circular include problem.